src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -191,6 +197,11 @@ public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, ImmutableA
 
         public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, bool? isNullableIfReferenceType)
         {
+            if ((object)typeSymbol == null)
+            {
+                return null;
```

Comment:
This is surprising. There are no similar checks in other factory methods above. They always return a new instance.

Reply:
Changed other factory methods to return `null` if `typeSymbol is null`.